src/appforms/src/core/060-RuleEngine.js

```diff
@@ -1282,6 +1282,18 @@
         }
       }
 
+      /**
+      * isSafeString - Checks if special characters in strings have already been escaped.
+      *
+      * @param  {string} str               The string to check.
+      * @return {boolean}
+      */
+      function isSafeString(str) {
```

Comment:
Looks like this returns true if one of the escaped chars is found in the string. What if the string was valid from the beginning? Then it will not have such charachters and `isSafeString` would return false if i read it correctly. Also i think it would be safer to check for the presence of unescaped charachters instead of escaped ones. A string is unsafe if it contains `<`, `>`, `&` etc.

Reply:
Yes ok that makes sense, I'll update it to check for unescaped characters instead

Reply:
Changing it to use to check for these characters `<`, `>`, `&`, will result in multiple escapes on the string because when escaped the string will look like `&lt;script&gt;console.log(&#x27;XSS:RADIO&#x27;)&lt;/script&gt;` and as this contains `&` it will be seen as `unsafe` and try escape it again

Reply:
We must use `escape` function on the `fieldValue` in order to match the same encoding that is being used on the form label for this to work.  We'll also have to check for escaped characters because if we check for unescaped characters the string will get escaped multiple times which is what we need to avoid